### PR TITLE
Add developer instructions to readme

### LINKED ISSUE #251: Script to download and build dependencies
Users who can't install packages are often missing dependencies needed to build Mosh from source, chiefly `protobuf`.  We could include a script which automatically downloads and builds dependencies.  We already [do this for Android](https://github.com/kmcallister/mosh/blob/android/android/build).

### PR DESCRIPTION
**Pull request by ahedberg**
Fixes #251

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,6 +147,22 @@ mosh-client packages to allow mosh-server usage without Perl.
 Notes for developers
 --------------------
 
+To start contributing to Mosh, install the following dependencies:
```
Comment (achernya) [inline README.md:150]: Can you mention that this is for a Debian or Ubuntu machine, possibly on  WSL?

File: README.md [hunk 2]
```diff
@@ -147,6 +147,22 @@ mosh-client packages to allow mosh-server usage without Perl.
...
+
+```
+$ sudo apt install -y protobuf-compiler libprotobuf-dev libutempter-dev autoconf automake nettle-dev
+$ brew install protobuf automake
```
Comment (achernya) [inline README.md:154]: And split this out into a separate macOS instruction?

**Review by achernya — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ahedberg) [reply README.md:150]: Done

File: README.md (hunk 2 above)
Comment (ahedberg) [reply README.md]: Done

File: README.md [hunk 3]
```diff
@@ -149,9 +149,18 @@ Notes for developers
 
 To start contributing to Mosh, install the following dependencies:
 
+Debian, Windows Subsystem for Linux:
+
+```
+$ sudo apt install -y build-essential debhelper-compat protobuf-compiler \
```
Comment (achernya) [inline README.md:155]: I don't believe we need debhelper-compat outside of Debian packaging.

File: README.md [hunk 4]
```diff
@@ -149,9 +149,18 @@ Notes for developers
...
+    libprotobuf-dev pkg-config libutempter-dev zlib1g-dev libncurses5-dev \
+    libssl-dev bash-completion locales tmux less
```
Comment (achernya) [inline README.md:157]: You shouldn't need locales (and we should remove that from the debian/rules file)

File: README.md [hunk 5]
```diff
@@ -149,9 +149,18 @@ Notes for developers
...
+```
+
+MacOS:
+
 ```
-$ sudo apt install -y protobuf-compiler libprotobuf-dev libutempter-dev autoconf automake nettle-dev
-$ brew install protobuf automake
+$ brew install protobuf automake zlib openssl bash tmux less
```
Comment (achernya) [inline README.md:163]: I would limit this to what we have in the CI for now

**Review by achernya — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (ahedberg) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (ahedberg) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (ahedberg) [reply README.md]: Done

**Review by achernya — APPROVED** [review]

**Pull request merged**